### PR TITLE
Support `DROP PROCEDURE` statement

### LINKED ISSUE #1282: Add support for DROP PROCEDURE
https://github.com/sqlparser-rs/sqlparser-rs/blob/9d15f7e9a92a4ef36fcc6d3166fd0221a22a20e2/src/parser/mod.rs#L4217
drop procedure stmt is allowed in many sql client impl.
(https://dev.mysql.com/doc/refman/8.0/en/drop-procedure.html,
https://learn.microsoft.com/en-us/sql/t-sql/statements/drop-procedure-transact-sql?view=sql-server-ver16,
https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/DROP-PROCEDURE.html,
...)

### PR DESCRIPTION
**Pull request by LorrensP-2158466**
Potentially closes #1282 

This introduces a new statement: `DropProcedure`. It has the same fields as `DropFunction` because it also has the same syntax.

The other way i could have implemented this was using the `Drop` statement and introducing a new `ObjectType` called `ObjectType::Procedure`, but because `DROP PROCEDURE ...` is the same (just different word) as `DROP FUNCTION...`, I thought this was weird to do.

I also added a test case in the same way as `DropFunction`.

Let me know what you guys think of this!

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -5867,12 +5893,12 @@ impl fmt::Display for DropFunctionOption {
 #[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
-pub struct DropFunctionDesc {
+pub struct DropFuncDesc {
```
Comment (iffyio) [inline src/ast/mod.rs:5896]: was it a specific reason to rename `DropFunctionDesc` to `DropFuncDesc`? if not we can probably undo the change given its a breaking change and the benefit is arguable, (and we have other similarly named objects like `DropFunctionOption`, `parse_drop_function_desc`)

File: src/parser/mod.rs [hunk 2]
```diff
@@ -4529,7 +4531,26 @@ impl<'a> Parser<'a> {
...
+    /// DROP PROCEDURE [ IF EXISTS ] name [ ( [ [ argmode ] [ argname ] argtype [, ...] ] ) ] [, ...]
+    /// [ CASCADE | RESTRICT ]
+    /// ```
+    fn parse_drop_procedure(&mut self) -> Result<Statement, ParserError> {
+        let if_exists = self.parse_keywords(&[Keyword::IF, Keyword::EXISTS]);
+        let proc_desc = self.parse_comma_separated(Parser::parse_drop_function_desc)?;
+        let option = match self.parse_one_of_keywords(&[Keyword::CASCADE, Keyword::RESTRICT]) {
+            Some(Keyword::CASCADE) => Some(ReferentialAction::Cascade),
+            Some(Keyword::RESTRICT) => Some(ReferentialAction::Restrict),
+            _ => None,
```
Comment (iffyio) [inline src/parser/mod.rs:4544]:
```suggestion
            Some(invalid) => return parser_err!(...)
            None => None,
```
Should this rather return an error if the action is unexpected?
seems otherwise e.g. `DROP PROCEDURE testproc SET NULL` will be accepted by the parser but it will silently discard the `SET NULL` part (we can also include a test case demonstrating the behavior)

File: src/ast/mod.rs (hunk 1 above)
Comment (LorrensP-2158466) [reply src/ast/mod.rs:5896]: Ah yeah you're right. I playing around with some new names since DropProcedure uses it. I'll change it back.

Comment (alamb) [reply src/ast/mod.rs]: done in bb0dad8

File: src/parser/mod.rs (hunk 2 above)
Comment (alamb) [reply src/parser/mod.rs]: I think they way that parse_one_of_keywords works, it will only return CASCADE or RESTRICT here (never Some(..)). 

I added a test in e782bf5 and changed the code to try and make this clearer

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @LorrensP-2158466  and @iffyio

**Comment by LorrensP-2158466** [comment]
Thanks for doing those final changes! It's very busy at the moment, sorry.

**Comment by alamb** [comment]
> Thanks for doing those final changes! It's very busy at the moment, sorry.

No problems! The best software in my opinion is a good team effort. 🚀
